core/src/main/java/hudson/EnvVars.java

```diff
@@ -388,6 +388,15 @@ public void putIfNotNull(String key, String value) {
         if (value!=null)
             put(key,value);
     }
+
+    /**
+     * Add entire map but filter null values out.
+     * @since TODO
+     */
+    public void putAllNonNull(Map<String, String> map) {
```

Comment:
Name inconsistent with `#putIfNotNull`. Perhaps the following?

```suggestion
    public void putAllIfNotNull(Map<String, String> map) {
```

Reply:
@daniel-beck putAllIfNotNull would suggest that put all values only if they are not null wouldn't it? That's why I kept it as putAllNonNull (so as to mean the non null values in map are added)